### PR TITLE
fix(install/update): replace remane by copy + delete

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

Replace rename PHP function by copy + delete to allow to move install directory from a device (partition) to another one

### CODE REVIEW

File: www/install/step_upgrade/process/process_step5.php [hunk 1]
```diff
@@ -52,6 +54,14 @@
 
 $name = 'install-' . $_SESSION['CURRENT_VERSION'] . '-' . date('Ymd_His');
 $completeName = _CENTREON_VARLIB_ . '/installs/' . $name;
-@rename(str_replace('step_upgrade', '', realpath(dirname(__FILE__) .'/../')), $completeName);
+$sourceInstallDir = str_replace('step_upgrade', '', realpath(dirname(__FILE__) .'/../'));
+
+try {
+    if (copy($sourceInstallDir, $completeName)) {
+        unlink();
```
Comment (jdelpierre) [inline www/install/step_upgrade/process/process_step5.php]: unlink($sourceInstallDir);

File: www/install/steps/process/process_step9.php [hunk 2]
```diff
@@ -54,9 +54,13 @@
...
         . '/install-' . $version . '-' . date('Ymd_His');
     $installDir = realpath(__DIR__ . '/../..');
-    $dependencyInjector['filesystem']->rename($installDir, $backupDir);
+    $dependencyInjector['filesystem']->copy($installDir, $backupDir);
+    $dependencyInjector['filesystem']->remove($installDir);
     if ($dependencyInjector['filesystem']->exists($installDir)) {
-        throw new \Exception('Cannot move directory from ' . $installDir . ' to ' . $backupDir);
+        throw new \Exception(
+            'Cannot move directory from ' . $installDir . ' to ' . $backupDir
+            . ', please remove it manually.'
```
Comment (kduret) [inline www/install/steps/process/process_step9.php]:
```suggestion
            . ', please move it manually.'
```

File: www/install/step_upgrade/process/process_step5.php [hunk 3]
```diff
@@ -35,8 +35,18 @@
 
 session_start();
 require_once __DIR__ . '/../../../../bootstrap.php';
+require_once '../../steps/functions.php';
+
+function recurseRmdir($dir) {
+    $files = array_diff(scandir($dir), array('.','..'));
```
Comment (kduret) [inline www/install/step_upgrade/process/process_step5.php]:
```suggestion
    $files = array_diff(scandir($dir), array('.', '..'));
```

File: www/install/step_upgrade/process/process_step5.php [hunk 4]
```diff
@@ -52,6 +62,14 @@
 
 $name = 'install-' . $_SESSION['CURRENT_VERSION'] . '-' . date('Ymd_His');
 $completeName = _CENTREON_VARLIB_ . '/installs/' . $name;
-@rename(str_replace('step_upgrade', '', realpath(dirname(__FILE__) .'/../')), $completeName);
+$sourceInstallDir = str_replace('step_upgrade', '', realpath(dirname(__FILE__) .'/../'));
```
Comment (kduret) [inline www/install/step_upgrade/process/process_step5.php]:
```suggestion
$sourceInstallDir = str_replace('step_upgrade', '', realpath(dirname(__FILE__) . '/../'));
```

File: www/install/step_upgrade/process/process_step5.php [hunk 5]
```diff
@@ -35,8 +35,45 @@
 
 session_start();
 require_once __DIR__ . '/../../../../bootstrap.php';
+require_once '../../steps/functions.php';
+
+function recurseRmdir($dir) {
```
Comment (ganoze) [inline www/install/step_upgrade/process/process_step5.php]:
```suggestion
function recurseRmdir($dir)
{
```

**Pull request merged**